PrintMap now uses spplot to create maps.

RTbecard opened the pull request:
- Scaling issue resolved
- Added `res` and `size` options so users can now save `png` figures which are larger than their screen.
- Added myself as an extra author.
- Presence absence legend is now on the interior top left to avoid overlapping with axis boundaries.
- Updated the roxygen comments with new parameters, but did not rebuild any documentation.
- Borders now fit to the region extent, so no extra white space within figure
- N and E units plotted on axis

![2016_04_02-1416](https://cloud.githubusercontent.com/assets/6226719/14226414/c352fb86-f8e1-11e5-8053-6ae3450f5497.png)

RTbecard commented:
Should I be worried about these failed checks?  I tested this on my machine by using `debugSource` to pass the modified `PrintMap` as a local function to the workflows for testing.

goldingn commented:
Those are errors (meaning tests failed to run) rather than fails (meaning tests found something wrong). We get these periodically, normally due to to internet connection issues on Travis (lots of modules & data are downloaded to test. The nightly builds failed recently too.

I'll manually trigger a new run now.

goldingn commented:
Would that it were so simple. Something else is up with the builds: https://github.com/zoonproject/zoon/issues/255

Will need to resolve that before we can merge. Hopefully won't take too long though!

AugustT commented:
@RTbecard This issue should be cleaerd up now. Can you pull the new master version from here and merge with your fork? When you push your new fork that should update the PR and start the checks off again.

AugustT commented:
Hold that thought...

AugustT commented:
Since you have only changed to one file I think I can pull this straight in now

RTbecard commented:
hmm... maybe hold off on the pull request for a bit... looking at the image I posted above, it looks like I might have mixed up the presence/absence colors and/or legend in the script!

Mind waiting till I get home this evening to double check the code?

AugustT commented:
Just reviewed your code, this looks really good, some nice attention to detail which makes the plot look really snappy. Will hold off on pulling in as I have spotted a couple of minor points.

- Line 75 will always be yes as you are inside an if statement with the same logical test.
- If `plot = FALSE` and `dir` is not `NULL` I think there might be a problem as the `plot.object` (created on line 61) wont exist at line 106.

RTbecard commented:
Fixed the minor points you mentioned.  I'm also adding support for plotting abundance points too, but I have to head to class right now so will finish later tonight hopefully.  Will let u know when I push the changes!

RTbecard commented:
Just updated my branch.
- Fixed Presence/Absence color reversal
- Added support for abundance datapoints
- Condensed code for creating enumerated filenames for outputs with same date/time
- Fixed 2 bugs pointed out by @AugustT 

PS.  How are you guys debugging modules?  Right now I'm just copying a module file into my `zoon` project, then using `sourceDebug` to load it as a local function so the workflow uses this instead of pulling from online.  I then just copy and paste my new module back into the `zoon - Modules` project.  However, I think this is causing messy commits.  Do you have a better workflow for debugging Modules?

![image](https://cloud.githubusercontent.com/assets/6226719/14492188/fbb75388-017d-11e6-84cc-432584956edd.png)

AugustT commented:
@RTbecard Have you looked at the building a module vignette? There I show how I go about building and testing modules

AugustT commented:
It is an iterative process that uses the BuildModule function. This is really how you should be doing things. For example building modules using BuildModule ensures that all your roxygen tags are in the right format. In your case, for example, each module can only have one email address associated with it whereas yours has 2.

RTbecard commented:
ooh!  I didn't realize that `BuildModule` did checks!  I assumed it directly passed the arguments into roxygen comments.  Will reread the vignette again.

AugustT commented:
Yeah, buildmodule is basically a way for ensuring that all the formatting is correct and it does basic checks. For more advanced checks you can run the tests in the modules 'package'. If you are using rstudio this is simple. In your copy of the modules package add your script to the others in teh R folder and run tests (ctrl+sft+T). This will test all of the modules in the R folder. These are teh same tests that we are going to run on the website when people submit modules (think kinda like CRAN checks)

AugustT commented:
@RTbecard Can you change

```
#' @author ZOON Developers, \email{zoonproject@@gmail.com}; James Campbell, \email{jamesadamcampbell@@gmail.com}
```

to 
```
#' @author ZOON Developers, James Campbell, \email{zoonproject@@gmail.com}
```

Just so that we only have the one email address there

AugustT commented:
To move this along I merged the pull request and then made the change myself.

RTbecard commented:
Thanks for taking care of this for me, and sorry for the delay!  I just moved into a new apartment last week and I still don't have internet on my main machine right now.

AugustT commented:
No problem!
